pkg/receive/writer.go

```diff
@@ -90,6 +94,33 @@ func (r *Writer) Write(ctx context.Context, tenantID string, wreq *prompb.WriteR
 				level.Debug(r.logger).Log("msg", "Out of bounds metric", "lset", lset, "sample", s)
 			}
 		}
+
+		// Current implemetation of app.AppendExemplar doesn't create a new series, so it must be already present.
+		// We drop the exemplars in case the series doesn't exist.
+		if ref != 0 && len(t.Exemplars) > 0 {
+			for _, ex := range t.Exemplars {
+				exLset := labelpb.ZLabelsToPromLabels(ex.Labels)
+				logger := log.With(r.logger, "exemplarLset", exLset, "exemplar", ex.String())
+
+				_, err = app.AppendExemplar(ref, lset, exemplar.Exemplar{
+					Labels: exLset,
+					Value:  ex.Value,
+					Ts:     ex.Timestamp,
+					HasTs:  true,
+				})
+				switch err {
+				case storage.ErrOutOfOrderExemplar:
+					numExemplarsOutOfOrder++
+					level.Debug(logger).Log("msg", "Out of order exemplar")
+				case storage.ErrDuplicateExemplar:
+					numExemplarsDuplicate++
+					level.Debug(logger).Log("msg", "Out of order exemplar")
+				case storage.ErrExemplarLabelLength:
+					numExemplarsLabelLength++
+					level.Debug(logger).Log("msg", "Label length for exemplar exceeds max limit")
```

Comment:
Possible to add length and the limit values? Provides value :)

Reply:
I have added limit but adding the current length is a bit tricky (we have loop over all the label names and values) so I am not sure if the overhead is worth it.